src/betterproto/__init__.py

```diff
@@ -691,17 +704,18 @@ def __repr__(self) -> str:
     if not TYPE_CHECKING:
 
         def __getattribute__(self, name: str) -> Any:
-            """
-            Lazily initialize default values to avoid infinite recursion for recursive
-            message types
-            """
-            value = super().__getattribute__(name)
-            if value is not PLACEHOLDER:
-                return value
+            # Raise `AttributeError` on attempts to access unset `oneof` fields
+            try:
+                group_current = super().__getattribute__("_group_current")
+            except AttributeError:
+                pass
+            else:
+                if name not in ("__class__", "_betterproto"):
+                    group = self._betterproto.oneof_group_by_field.get(name)
+                    if group is not None and group_current[group] != name:
+                        raise AttributeError()
```

Comment:
It may be useful to add context to the missing attribute to the exception message.

The common attribute error that Python raises is as follows:
```
raise AttributeError(f"'{self.__class__.__name__}' object has no attribute '{name}'")
```
However, you may want to give it even more context:
```
raise AttributeError(f"'{self.__class__.__name__}.{group}' is set to '{group_current[group]}' not '{name}'")
```
You may want to use `str.format` rather than an f-string depending on which Python versions this project supports.